cocotb/share/makefiles/Makefile.pylib.Darwin

```diff
@@ -29,12 +29,12 @@
 
 # All common python related rules
 
+_PYTHON_BIN?=$(shell cocotb-config --python-bin)
 ifneq ($(COCOTB_PYTHON_DEBUG),)
```

Comment:
python-dbg packages are special to (at least) Debian in that they are built with `--with-pydebug`, but use the same site-packages as the non-dbg variant. From the README that Debian installs:

```
Using the python-dbg builds
---------------------------

 * Call the python-dbg or the pythonX.Y-dbg binaries instead of the
   python or pythonX.Y binaries.

 * Properties of the debug build are described in
   /usr/share/doc/@PVER@/SpecialBuilds.txt.gz.
   The debug interpreter is built with Py_DEBUG defined.

 * From SpecialBuilds.txt: This is what is generally meant by "a debug
   build" of Python.  Py_DEBUG implies LLTRACE, Py_REF_DEBUG,
   Py_TRACE_REFS, and PYMALLOC_DEBUG (if WITH_PYMALLOC is enabled).
   In addition, C assert()s are enabled (via the C way: by not defining
   NDEBUG), and some routines do additional sanity checks inside
   "#ifdef Py_DEBUG" blocks.
```

This code was added in 11c504e6 probably as debugging aid (the mentioned issue doesn't talk about the debug flag).

I'm inclined to just remove this functionality since it
- Is not documented.
- Can be easily emulated by setting PYTHON_BIN to the `-dbg` variant.

@stuarthodgson since you added that code in the first place, does that sound OK to you?